include/ibamr/WaveDampingStrategy.h

```diff
@@ -0,0 +1,125 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+/////////////////////// INCLUDE GUARD ////////////////////////////////////
+
+#ifndef included_WaveDampingStrategy
+#define included_WaveDampingStrategy
+
+///////////////////////////// INCLUDES ///////////////////////////////////
+#include <string>
+
+#include "Variable.h"
+#include "ibamr/AdvDiffHierarchyIntegrator.h"
+#include "ibamr/INSVCStaggeredHierarchyIntegrator.h"
+#include "ibtk/ibtk_utilities.h"
+#include "tbox/Pointer.h"
+
+namespace IBAMR
+{
+/*!
+ * A collection of post processing call back functions to be hooked into
+ * IBAMR::HierarchyIntegrator class to employ a wave damping zone of some
+ * prescribed length within the computational domain. Additional strategies
+ * can be implemented as simple C functions within this collection.
+ *
+ * The instance of a WaveDampingStrategy and the particular damping function
+ * should be registered with an appropriate hierarchy integrator via
+ * registerPostprocessIntegrateHierarchyCallback.
+ *
+ * \param ctx is the pointer to WaveDampingData struct.
+ */
+
+namespace WaveDampingFunctions
+{
+void callRelaxationZoneCallbackFunction(double current_time,
+                                        double new_time,
+                                        bool skip_synchronize_new_state_data,
+                                        int num_cycles,
+                                        void* ctx);
+
+void callConservedWaveAbsorbingCallbackFunction(double current_time,
+                                                double new_time,
+                                                bool skip_synchronize_new_state_data,
+                                                int num_cycles,
+                                                void* ctx);
+} // namespace WaveDampingFunctions
+
+/*!
+ * A struct holding the required information used by the wave damping function.
+ */
+struct WaveDampingData
+{
+    /*
+     * Pointers to the fluid and advection-diffusion integrators.
+     */
+    SAMRAI::tbox::Pointer<IBAMR::INSVCStaggeredHierarchyIntegrator> d_ins_hier_integrator;
+    SAMRAI::tbox::Pointer<IBAMR::AdvDiffHierarchyIntegrator> d_adv_diff_hier_integrator;
+
+    /*
+     * Pointer to the level set variable representing the wave interface.
+     */
+    SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > d_phi_var;
+
+    /*
+     *  Pointer to phi variable's new context.
+     *  \note We modify the phi value after the end of each timestep.
+     */
+    SAMRAI::tbox::Pointer<SAMRAI::hier::VariableContext> d_phi_new_ctx;
+
+    /*
+     * Start and end coordinates of the damping zone, water depth, and damping coefficient.
+     */
+    double d_x_zone_start, d_x_zone_end, d_depth;
+    double d_alpha;
+    int d_sign_gas_phase = 1;
+};
+
+/*!
+ * A struct holding the required information used by the variable alpha damping method
```

Comment:
Can you add a brief comment saying what `operator()` does here?

Also, is there a reason to prefer functors over plain-old-functions + `Data` object? It seems like most things are being specified through static functions + `ctx`.

(I am OK with functors but also prefer that users interact with the classes in a consistent way when possible.)

Reply:
We are using a functor as required by boost's Newton-Ralphson root finding interface
https://www.boost.org/doc/libs/1_62_0/libs/math/doc/html/math_toolkit/roots/roots_deriv.html

Reply:
Got it. Does this need to be part of the "public" interface.

Reply:
I don't think so. I think we can move this `struct` to anonymous namespace of `WaveDampingStrategy.cpp` file. Do you agree?

Reply:
Sounds great!